ecl/eclcc/WORKUNITS.rst

```diff
@@ -0,0 +1,862 @@
+         <att name="_fileName" value="names"/>
+        </node>
+        <node id="3" label="Output&#10;Result #1">
+         <att name="definition" value="workuniteg1.ecl(4,1)"/>
+         <att name="_kind" value="16"/>
+         <att name="ecl" value="OUTPUT(..., workunit);&#10;"/>
+         <att name="recordSize" value="120"/>
+        </node>
+       </graph>
+      </att>
+     </node>
+    </graph>
+   </xgmml>
+  </Graph>
+  
+The xml for each activity might contain the following information:
+
+* A unique id (e.g. id="2").
+* The "kind" of the activity e.g., <att name="_kind" value="77"/>.  The value is an item from the enum ThorActivityKind in eclhelper.hpp.
+* The ECL that created the activity. E.g., <att name="ecl" value="...">
+* Name of the original definition.  E.g., <att name="name" value="results"/>
+* Location (e.g., file, line number, column) of the original ECL. E.g., <att name="definition" value="workuniteg1.ecl(3,1)"/>
+* Meta information the code generator has deduced about the activity output.  Examples include the 
+  record size, expected number of rows, sort order etc.  E.g., <att name="recordSize" value="120"/>
+* Hints, which are used for fine control of options for a particular activity (e.g,, the number of threads to use while sorting).
+* Record counts and stats once the job has executed.  (These are logically associated with the activities in the graph, but stored separately.)
+
+Graphs also contain edges which can take one of 3 forms:
+
+Edges within graphs
+    These are used to indicate how the activities are connected.  The source activity is used as the input to the target activity.
+    These edges have the following format::
+    
+        <edge id="<source-activity-id>_<output-count>" source="<source-activity-id>" target="<target-activity-id">
+    
+    There is only one edge in our example workunit: <edge id="2_0" source="2" target="3"/>.
+    
+Edges between graphs
+    These are used to indicate direct dependencies between activities.  For instance there will be an edge connecting the activity that
+    writes a spill file to the activity that reads it.  These edges have the following format::
+
+        <edge id="<source-activity-id>_<target-activity-id>" source="<source-subgraph-id>" target="<target-subgraph-id>"
+           <att name="_sourceActivity" value="<source-activity-id>"/>
+           <att name="_targetActivity" value="<target-activity-id>"/>
+        </edge>
+    
+    Roxie often optimizes spilled datasets and treats these edges as equivalent to the edges between activities.
+        
+Other dependencies.
+    These are similar to the edges between graphs, but they are used for values that are used within an activity.  For
+    instance one part of the graph may calculate the maximum value of a column, and another activity may filter out
+    all records that don't match that maximum value.  The format is the same as the edges between graphs except the edge contains
```

Comment:
don't -> do not (again should decide on formality of this doc?)
except **_that_** the